cmd/analyzer/main.go

```diff
@@ -25,25 +25,50 @@ const (
 )
 
 var (
-	settingsFile      = flag.String("provider-settings", "provider_settings.json", "path to the provider settings")
-	rulesFile         = flag.StringArray("rules", []string{"rule-example.yaml"}, "filename or directory containing rule files")
-	outputViolations  = flag.String("output-file", "output.yaml", "filepath to to store rule violations")
-	errorOnViolations = flag.Bool("error-on-violation", false, "exit with 3 if any violation are found will also print violations to console")
-	labelSelector     = flag.String("label-selector", "", "an expression to select rules based on labels")
-	depLabelSelector  = flag.String("dep-label-selector", "", "an expression to select dependencies based on labels. This will filter out the violations from these dependencies as well these dependencies when matching dependency conditions.")
-	logLevel          = flag.Int("verbose", 9, "level for logging output")
-	enableJaeger      = flag.Bool("enable-jaeger", false, "enable tracer exports to jaeger endpoint")
-	jaegerEndpoint    = flag.String("jaeger-endpoint", "http://localhost:14268/api/traces", "jaeger endpoint to collect tracing data")
-	limitIncidents    = flag.Int("limit-incidents", 1500, "Set this to the limit incidents that a given rule can give, zero means no limit")
-	limitCodeSnips    = flag.Int("limit-code-snips", 20, "limit the number code snippets that are retrieved for a file while evaluating a rule, 0 means no limit")
-	analysisMode      = flag.String("analysis-mode", "", "select one of full or source-only to tell the providers what to analyize. This can be given on a per provider setting, but this flag will override")
-	noDependencyRules = flag.Bool("no-dependency-rules", false, "Disable dependency analysis rules")
+	settingsFile      string
+	rulesFile         []string
+	outputViolations  string
+	errorOnViolations bool
+	labelSelector     string
+	depLabelSelector  string
+	logLevel          int
+	enableJaeger      bool
+	jaegerEndpoint    string
+	limitIncidents    int
+	limitCodeSnips    int
+	analysisMode      string
+	noDependencyRules bool
+
+	rootCmd = &cobra.Command{
+		Use:   "analyze",
+		Short: "Tool for working with analyzer-lsp",
+		Run:   func(c *cobra.Command, args []string) {},
+	}
 )
 
+func init() {
+	rootCmd.PersistentFlags().StringVar(&settingsFile, "settings-file", "provider_settings.json", "path to the provider settings")
```

Comment:
I wonder do these need to be persistent for all potential sub commands?

Reply:
You're right. I don't think any of them needed to be persistent actually. Updated